src/test/javascript/spec/component/grading-system/grading-system.component.spec.ts

```diff
@@ -0,0 +1,292 @@
+        );
+
+        comp.delete();
+
+        expect(gradingSystemDeleteForCourseStub).to.have.been.calledOnceWith(comp.courseId);
+        expect(comp.existingGradingScale).to.equal(false);
+    });
+
+    it('should delete grading scale for exam', () => {
+        comp.existingGradingScale = true;
+        comp.isExam = true;
+        const gradingSystemDeleteForExamStub = sinon.stub(gradingSystemService, 'deleteGradingScaleForExam').returns(
+            of(
+                new HttpResponse<{}>({ body: [] }),
+            ),
+        );
+
+        comp.delete();
+
+        expect(gradingSystemDeleteForExamStub).to.have.been.calledOnceWith(comp.courseId, comp.examId);
+        expect(comp.existingGradingScale).to.equal(false);
+    });
+
+    it('should create grading scale correctly for course', () => {
+        comp.existingGradingScale = false;
+        const gradingSystemCreateForCourseStub = sinon.stub(gradingSystemService, 'createGradingScaleForCourse').returns(
+            of(
+                new HttpResponse<GradingScale>({ body: comp.gradingScale }),
+            ),
+        );
+
+        comp.save();
+
+        expect(gradingSystemCreateForCourseStub).to.have.been.calledOnceWith(comp.courseId);
+        expect(comp.existingGradingScale).to.equal(true);
+    });
+
+    it('should create grading scale correctly for exam', () => {
+        comp.existingGradingScale = false;
+        comp.isExam = true;
+        const gradingSystemCreateForExamStub = sinon.stub(gradingSystemService, 'createGradingScaleForExam').returns(
+            of(
+                new HttpResponse<GradingScale>({ body: comp.gradingScale }),
+            ),
+        );
+
+        comp.save();
+
+        expect(gradingSystemCreateForExamStub).to.have.been.calledOnceWith(comp.courseId, comp.examId);
+        expect(comp.existingGradingScale).to.equal(true);
+    });
+
+    it('should update grading scale correctly for course', () => {
+        comp.existingGradingScale = true;
+        const gradingSystemUpdateForCourseStub = sinon.stub(gradingSystemService, 'updateGradingScaleForCourse').returns(
+            of(
+                new HttpResponse<GradingScale>({ body: comp.gradingScale }),
+            ),
+        );
+
+        comp.save();
+
+        expect(gradingSystemUpdateForCourseStub).to.have.been.calledOnceWith(comp.courseId);
+        expect(comp.existingGradingScale).to.equal(true);
+    });
+
+    it('should update grading scale correctly for exam', () => {
+        comp.existingGradingScale = true;
+        comp.isExam = true;
+        const gradingSystemUpdateForExabStub = sinon.stub(gradingSystemService, 'updateGradingScaleForExam').returns(
+            of(
+                new HttpResponse<GradingScale>({ body: comp.gradingScale }),
+            ),
+        );
+
+        comp.save();
+
+        expect(gradingSystemUpdateForExabStub).to.have.been.calledOnceWith(comp.courseId);
+        expect(comp.existingGradingScale).to.equal(true);
```

Comment:
How about the effects of data that is coming from the gradingSystemService what happens to it?